javascript/classes.js

```diff
@@ -0,0 +1,53 @@
+export default () => {
+    
+    class Character{
+        
+        constructor(x, y){
+            
+            Character.counter++;
+            this.x = Math.round(Math.random() * 10);
+            this.y = Math.round(Math.random() * 10);
+            if (this.constructor === Character){
+                throw new Error("Character cannot be instantianted");
+            }
+        }
+        // static get count(){
+        //     return this.hasOwnProperty('counter') ? this.counter: 0;
+        // }
+        setXY(x, y){
+            if (x > 10 || y > 10){
+                throw new Error("Cordinates out of bounds");
+            }else{
+                this.x = x;
+                this.y = y;
+            }
+        }
+        getXY(){
+            return [this.x, this.y];
+        }
+        
+    }
+    //staticka promenljiva
+    Character.counter = 0;
+
+    class PLayerCharacter extends Character{
+        constructor(){
+            super();
+        }
+    }
+    class NonPLayerCharacter extends Character{
+        constructor(){
+            super();
+        }
+    }
+    var player = new PLayerCharacter();
+    console.log(player);
+    console.log(player.getXY());
+    try{
+        player.setXY(11, 1);
+    }catch(err){
+        console.log(err);
+    }
+    var player2 = new NonPLayerCharacter();
+    console.log("Numnber of players", Character.counter);
```

Comment:
ovo bi sad trebalo ispisati sve preko `Prototypes`, ali ne moras raditi evo ti link pa procitaj samo https://medium.com/@parsyval/javascript-prototype-vs-class-a7015d5473b
Fora je u tome sto u JS ne postoje klase kao u drugim jezicima. To je napravljeno samo radi citljivosti i boljeg snalaznjea, pa se za klase kaze da su `syntactical sugar`